src/Languages/Editor/Test/Shell/TestShellBase.cs

```diff
@@ -39,6 +39,10 @@ public class TestShellBase {
             await UIThreadHelper.Instance.InvokeAsync(action);
         }
 
+        public async Task<TResult> DispatchOnMainThreadAsync<TResult>(Func<TResult> action, CancellationToken cancellationToken = new CancellationToken()) {
```

Comment:
Just different syntax. InvokeAsync allows to `await` its task and continue execution.